File: src/test/java/de/hdawg/rankinginfo/web/FederationControllerTest.java
package de.hdawg.rankinginfo.web;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class FederationControllerTest extends WebControllerTestBase {

  @Test
  @DisplayName("GET federations returns 200")
  void getFederationsReturns200() throws Exception {
    mockMvc.perform(get("/federations")).andExpect(status().isOk());
  }

  @Test
  @DisplayName("GET federations populates federations model")
  void getFederationsPopulatesFederationsModel() throws Exception {
    mockMvc.perform(get("/federations")).andExpect(status().isOk());
  }

  @Test
  @DisplayName("GET federations with empty DB returns empty map")
  void getFederationsWithEmptyDbReturnsEmptyMap() throws Exception {
    rankingRepository.deleteAll();
    mockMvc.perform(get("/federations")).andExpect(status().isOk());
  }
}

File: src/test/kotlin/de/hdawg/rankinginfo/web/FederationControllerTest.kt
package de.hdawg.rankinginfo.web

import org.junit.jupiter.api.Test
import org.springframework.test.web.servlet.get

class FederationControllerTest : WebControllerTestBase() {
    @Test
    fun `GET federations returns 200`() {
        mockMvc.get("/federations").andExpect { status { isOk() } }
    }

    @Test
    fun `GET federations populates federations model`() {
        mockMvc.get("/federations").andExpect {
        }
    }

    @Test
    fun `GET federations with empty DB returns empty map`() {
        rankingRepository.deleteAll()
        mockMvc.get("/federations").andExpect {
            status { isOk() }
        }
    }
}
